tools/src/main/java/org/apache/kafka/tools/reassign/ReassignPartitionsCommand.java

```diff
@@ -0,0 +1,1523 @@
+
+        long numRackless = results.stream().filter(m -> !m.rack.isPresent()).count();
+        if (enableRackAwareness && numRackless != 0 && numRackless != results.size()) {
+            throw new AdminOperationException("Not all brokers have rack information. Add " +
+                "--disable-rack-aware in command line to make replica assignment without rack " +
+                "information.");
+        }
+        return results;
+    }
+
+    /**
+     * Parse and validate data gathered from the command-line for --generate
+     * In particular, we parse the JSON and validate that duplicate brokers and
+     * topics don't appear.
+     *
+     * @param reassignmentJson       The JSON passed to --generate .
+     * @param brokerList             A list of brokers passed to --generate.
+     *
+     * @return                       A tuple of brokers to reassign, topics to reassign
+     */
+    static Tuple2<List<Integer>, List<String>> parseGenerateAssignmentArgs(String reassignmentJson,
+                                                                           String brokerList) throws JsonMappingException {
+        List<Integer> brokerListToReassign = Arrays.stream(brokerList.split(",")).map(Integer::parseInt).collect(Collectors.toList());
+        Set<Integer> duplicateReassignments = ToolsUtils.duplicates(brokerListToReassign);
+        if (!duplicateReassignments.isEmpty())
+            throw new AdminCommandFailedException(String.format("Broker list contains duplicate entries: %s", duplicateReassignments));
+        List<String> topicsToReassign = parseTopicsData(reassignmentJson);
+        Set<String> duplicateTopicsToReassign = ToolsUtils.duplicates(topicsToReassign);
+        if (!duplicateTopicsToReassign.isEmpty())
+            throw new AdminCommandFailedException(String.format("List of topics to reassign contains duplicate entries: %s",
+                duplicateTopicsToReassign));
+        return new Tuple2<>(brokerListToReassign, topicsToReassign);
+    }
+
+    /**
+     * The entry point for the --execute and --execute-additional commands.
+     *
+     * @param adminClient                 The AdminClient to use.
+     * @param additional                  Whether --additional was passed.
+     * @param reassignmentJson            The JSON string to use for the topics to reassign.
+     * @param interBrokerThrottle         The inter-broker throttle to use, or a negative
+     *                                    number to skip using a throttle.
+     * @param logDirThrottle              The replica log directory throttle to use, or a
+     *                                    negative number to skip using a throttle.
+     * @param timeoutMs                   The maximum time in ms to wait for log directory
+     *                                    replica assignment to begin.
+     * @param time                        The Time object to use.
+     */
+    static void executeAssignment(Admin adminClient,
```

Comment:
Original `ReassignPartitionsCommand#executeAssignment` has parameters with default values. Does we have to add set of `executeAssignment` overloaded methods? Does is make sense?